### PR TITLE
Add witness output to goto checker [blocks: 3583]

### PR DESCRIPTION
**Pull request by peterschrammel**
Based on #3564, only review last commit

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 2107223cb8908af2e378c2569b2923a4c3a7eeab).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/94898955
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

File: src/goto-checker/goto_checker.h [hunk 1]
```diff
@@ -59,6 +59,11 @@ class goto_checkert
   /// This builds and returns the counterexample
   virtual goto_tracet build_error_trace() const = 0;
 
+  /// This outputs an error witness
+  virtual void output_error_witness(const goto_tracet &) = 0;
+  /// This outputs an proof
```
Comment (tautschnig) [inline src/goto-checker/goto_checker.h]: I have (wrongly) commented on this commit elsewhere: s/an/a/

**Review by tautschnig — APPROVED** [review]

File: src/goto-checker/goto_checker.h [hunk 2]
```diff
@@ -59,6 +59,11 @@ class goto_checkert
   /// This builds and returns the counterexample
   virtual goto_tracet build_error_trace() const = 0;
 
+  /// This outputs an error witness
```
Comment (smowton) [inline src/goto-checker/goto_checker.h]: Drop Doxy comments that just repeat the function name? Or, provide a link to define witness?

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: db83fce28fef2a75148bef12b376138915bb7cf4).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97163305

**Pull request merged**